ister_gui.py

```diff
@@ -2146,6 +2167,12 @@ def handler(self, config):
                         else:
                             if bundle in config['Bundles']:
                                 config['Bundles'].remove(bundle)
+
+        # update only if user had chance to unselect
+        if not any(bundle['name'] == 'sysadmin-basic'
+                   for bundle in self.required_bundles):
+            self.sysadmin_basic_state = 'sysadmin-basic' in config['Bundles']
```

Comment:
config['Bundles'] holds the bundles that will be installed on the system, so all user selections plus required bundles. default_bundles are just selected by default, not included by default.

I don't add sysadmin-basic back into default_bundles (I don't include it as selected by default again) because I want it to be removed from the required_bundles list and placed into the optional bundles list in the same state it was in before it was *made* to be required...

So I track the state the sysadmin-basic bundle was in before being made required and if I have to put it back, I assign it the same state it was before requiring it. I acknowledge that is hard to follow.